src/index/src/fulltext_index/search/tantivy.rs

```diff
@@ -77,9 +79,48 @@ impl FulltextIndexSearcher for TantivyFulltextIndexSearcher {
         let query = query_parser
             .parse_query(query)
             .context(TantivyParserSnafu)?;
-        let docs = searcher
+        let doc_addrs = searcher
             .search(&query, &DocSetCollector)
             .context(TantivySnafu)?;
-        Ok(docs.into_iter().map(|d| d.doc_id).collect())
+
+        let seg_metas = self
+            .index
+            .searchable_segment_metas()
+            .context(TantivySnafu)?;
+
+        // FAST PATH: only one segment, the doc id is the same as the row id.
+        //            Also for compatibility with the old version.
+        if seg_metas.len() == 1 {
+            return Ok(doc_addrs.into_iter().map(|d| d.doc_id).collect());
+        }
+
+        // SLOW PATH: multiple segments, need to calculate the row id.
```

Comment:
If we limit the total rows in an SST file, can we get rid of this issue?

Reply:
It can alleviate the problem, but there is no guarantee